nima/http/media/jackson/src/main/java/io/helidon/nima/http/media/jackson/JacksonMediaSupportProvider.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.nima.http.media.jackson;
+
+import io.helidon.common.GenericType;
+import io.helidon.common.Weighted;
+import io.helidon.common.http.Headers;
+import io.helidon.common.http.Http;
+import io.helidon.common.http.HttpMediaType;
+import io.helidon.common.http.WritableHeaders;
+import io.helidon.common.media.type.MediaTypes;
+import io.helidon.nima.http.media.EntityReader;
+import io.helidon.nima.http.media.EntityWriter;
+import io.helidon.nima.http.media.MediaContext;
+import io.helidon.nima.http.media.spi.MediaSupportProvider;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+
+import static io.helidon.common.http.Http.HeaderValues.CONTENT_TYPE_JSON;
+
+/**
+ * {@link java.util.ServiceLoader} provider implementation for Jackson media support.
+ */
+@SuppressWarnings({"rawtypes", "unchecked"})
+public class JacksonMediaSupportProvider implements MediaSupportProvider, Weighted {
+    private JacksonReader reader;
+    private JacksonWriter writer;
+    private ObjectMapper objectMapper;
+
+    @Override
+    public void init(MediaContext context) {
+        objectMapper = new ObjectMapper()
+                .registerModule(new ParameterNamesModule())
+                .registerModule(new Jdk8Module())
+                .registerModule(new JavaTimeModule());
```

Comment:
Suppose I, as an end developer, wish to customize the `ObjectMapper` in use (a very common thing to do with Jackson). Is my only option to supply a different implementation of `MediaSupportProvider`?

Reply:
Right now, yes. We have an issue to separate the Provider from Support, then we can do a custom instance.

Reply:
https://github.com/helidon-io/helidon/issues/6421